app/connectors/GuaranteeBalanceConnector.scala

```diff
@@ -48,7 +52,7 @@ class GuaranteeBalanceConnector @Inject() (http: HttpClient, appConfig: Frontend
   )
 
   def submitBalanceRequest(request: BalanceRequest)(implicit hc: HeaderCarrier): Future[Either[HttpResponse, BalanceRequestResponse]] = {
-    val url = s"${appConfig.guaranteeBalanceUrl}/balances"
+    val url = new URL(s"${appConfig.guaranteeBalanceUrl}/balances")
```

Comment:
There's a url interpolator built in. Try changing this to
`val url = url"${appConfig.guaranteeBalanceUrl}/balances"`
You may need to import it